sentinel/src/lib/muting_utils.js

```diff
@@ -122,20 +164,21 @@ const updateMuteState = (contact, muted) => {
     return batches
       .reduce((promise, batch) => {
         return promise
-          .then(() => getContactsAndSubjectIds(batch))
+          .then(() => getContactsAndSubjectIds(batch, muted))
           .then(result => Promise.all([
-            updateContacts(result.contacts, muted ? timestamp : false),
-            updateRegistrations(result.subjectIds, muted)
+            updateContacts(result.contacts, muted),
+            updateRegistrations(result.subjectIds, muted),
+            updateMutingHistories(result.contacts, muted, reportId)
           ]));
       }, Promise.resolve());
   });
 };
 
-const isMutedInLineage = doc => {
+const isMutedInLineage = (doc, returnId) => {
   let parent = doc && doc.parent;
   while (parent) {
     if (parent.muted) {
-      return true;
+      return returnId ? parent._id : true;
```

Comment:
Why not always return the parent._id and rely on it being truthy?

Reply:
Done! I thought about it too.